### PR TITLE
Groups feature

### LINKED ISSUE #72: As a Visitor, I want to be able to create a group of other Visitors, so that I can use all the interraction features related to groups
- [x] #94 
- [ ] #165 
- [ ] #163 
- [ ] #170  
- [ ] #172

### PR DESCRIPTION
**Pull request by maltherd**
Closes #72

### CODE REVIEW

**Comment by maltherd** [comment]
Right, the tests fail because User's age field has been made a Long now.
That's because, for some reason, Firebase wants to store numbers as Longs.
I'll fix those tests

**Comment by maltherd** [comment]
ha now it's a UI thing.
the event edit page test probably does not handle the groups button being at the same spot ? idk

**Comment by choelzl** [comment]
> ha now it's a UI thing.
> the event edit page test probably does not handle the groups button being at the same spot ? idk

Some times there are still UI-timeouts, a quick restart solves the issue

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,6 +4,10 @@
     xmlns:tools="http://schemas.android.com/tools"
     package="ch.epfl.polycrowd">
 
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
```
Comment (choelzl) [inline app/src/main/AndroidManifest.xml:8]: Duplicated permissions ?

File: app/src/main/java/ch/epfl/polycrowd/firebase/FirebaseInterface.java [hunk 2]
```diff
@@ -186,6 +195,156 @@ public void getEventById(String eventId, EventHandler eventHandler) {
                     }).addOnFailureListener(e -> Log.e(TAG, "Error retrieving document with id " + eventId));
     }
 
+    @RequiresApi(api = Build.VERSION_CODES.O)
```
[the remark this answers is not in the document]
Comment (choelzl) [reply app/src/main/java/ch/epfl/polycrowd/firebase/FirebaseInterface.java:211]: I agree with cc

File: app/src/test/java/ch/epfl/polycrowd/UserTest.java [hunk 3]
```diff
@@ -42,7 +42,7 @@ public void testGetFromDocument(){
         Map<String, Object> data = new HashMap<>();
         data.put("username", "test");
         data.put("email" , "test@test");
-        data.put("age" , 3);
+        data.put("age" , 3L);
```
Comment (choelzl) [inline app/src/test/java/ch/epfl/polycrowd/UserTest.java]: ??

**Review by choelzl — COMMENTED** [review]
Unit tests at least for `logic/groups` would have been nice

**Review by SachaKor — APPROVED** [review]

File: app/src/test/java/ch/epfl/polycrowd/UserTest.java (hunk 3 above)
Comment (maltherd) [reply app/src/test/java/ch/epfl/polycrowd/UserTest.java]: Firebase utilise des Long pour les nombres => la fonction getFromDocuments() prend ça comme type. 
Java c'est relou donc ça veut pas juste caster un 3 en long donc il faut le déclarer avec une litérale long => 3L

**Review by choelzl — APPROVED** [review]

**Pull request merged**